src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20220914155611_EES3695MakePublicationsContactIdNonnullable.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Migrations.ContentMigrations
+{
+    public partial class EES3695MakePublicationsContactIdNonnullable : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropForeignKey(
+                name: "FK_Publications_Contacts_ContactId",
+                table: "Publications");
+
+            migrationBuilder.AlterColumn<Guid>(
+                name: "ContactId",
+                table: "Publications",
+                type: "uniqueidentifier",
+                nullable: false,
+                defaultValue: new Guid("00000000-0000-0000-0000-000000000000"),
+                oldClrType: typeof(Guid),
+                oldType: "uniqueidentifier",
+                oldNullable: true);
+
+            migrationBuilder.AddForeignKey(
+                name: "FK_Publications_Contacts_ContactId",
+                table: "Publications",
+                column: "ContactId",
+                principalTable: "Contacts",
+                principalColumn: "Id",
+                onDelete: ReferentialAction.Cascade);
```

Comment:
Not 100% sure on this, but I think this is telling EF that if a Contact record is deleted, it'd then try to cascade that and delete its owning Publication too.  Worth checking with other devs?

Reply:
Assuming that's so, would suggest changing it to ReferentialAction.Restrict.  This'll mean EF would not allow the deletion of a Contact that is in use by a Publication.

Reply:
I haven't tested it, but I think you're right. I've changed it to `Restrict`.

Reply:
Ok Ben said that this actually means if a Publication gets deleted, the associated Contact will be deleted. We think this is fine so I've changed this back to Cascade. I will test this to make sure it's right

Reply:
We've had a discussion on Slack - @markhiveit to test this out to be sure of the correct way round.

Reply:
Ignore what I said now, I was thinking this FK was being added to the contacts table, but it's a FK on the publications table.

So it does want to be `onDelete: ReferentialAction.Restrict` but we are left with a potential problem of leaving orphaned contacts if publications are deleted.

Reply:
Tested this.

`DELETE  FROM Contacts WHERE Id = '35424c89-7ee1-4951-b1e8-f786c69e137c';` also deleted the publication
`DELETE FROM Publications WHERE ContactId = '35424c89-7ee1-4951-b1e8-f786c69e137c';` didn't result in the Contact being deleted.

I'll change the ReferentialAction to Restrict.

Reply:
Switched to Restrict and reset the test data locally to reapply the migration. Tried to delete a contact and it wouldn't let me while the associated publication still existed.